integrations/integrations.go

```diff
@@ -0,0 +1,232 @@
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	env *environment.Env,
+	svrCtx *server.Context,
+	_ serverconfig.Config, //nolint:gocritic // hugeParam
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	g, ctx := getCtx(svrCtx, true)
+
+	svrCtx.Logger.Info("starting Monomer node in-process")
+	// --- Start the Monomer node ---
+	wrappedApp := &WrappedApplication{app}
+	_, err := startMonomerNode(wrappedApp, env, svrCtx)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node. The
+// `MonomerGenesisPath` flag must be set in viper before invoking this.
+func startMonomerNode(wrappedApp *WrappedApplication, env *environment.Env, svrCtx *server.Context) (*node.Node, error) {
+	// --- Bootstrap listeners and DBs ---
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+	cometListener, err := net.Listen("tcp", cmtListenAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	appdb := dbm.NewMemDB()
+	blockdb := dbm.NewMemDB()
+	txdb := cometdb.NewMemDB()
+	mempooldb := dbm.NewMemDB()
+
+	// --- Load Monomer genesis doc ---
+	monomerGenesisPath := viper.GetString("monomer-genesis-path")
+	svrCtx.Logger.Info("loading Monomer genesis from", "path", monomerGenesisPath)
+
+	appGenesis, err := genutiltypes.AppGenesisFromFile(monomerGenesisPath)
+	if err != nil {
+		svrCtx.Logger.Error("failed to load Monomer genesis", "error", err)
+		return nil, err
+	}
+
+	var appState map[string]json.RawMessage
+	if err := json.Unmarshal(appGenesis.AppState, &appState); err != nil {
+		svrCtx.Logger.Error("failed to unmarshal app state", "error", err)
+		return nil, fmt.Errorf("failed to unmarshal app state: %w", err)
+	}
+
+	// --- Create a Monomer node ---
+	n := node.New(
+		wrappedApp,
+		&genesis.Genesis{
+			ChainID:  chainID,
+			AppState: appState,
+		},
+		engineWS,
+		cometListener,
+		blockdb,
+		mempooldb,
+		txdb,
+		&node.SelectiveListener{
+			OnEngineHTTPServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve engine HTTP", "error", err)
+			},
+			OnEngineWebsocketServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve engine websocket", "error", err)
+			},
+			OnCometServeErrCb: func(err error) {
+				svrCtx.Logger.Error("failed to serve CometBFT", "error", err)
+			},
+		},
+	)
+	svrCtx.Logger.Info("spinning up Monomer node")
+
+	nodeCtx, nodeCtxCancel := context.WithCancel(context.Background())
```

Comment:
Let's defer the cancel here instead of waiting til the end of the function
```suggestion
	nodeCtx, nodeCtxCancel := context.WithCancel(context.Background())
	env.Defer(nodeCtxCancel)
```